### PR TITLE
MAISTRA-2172 Return error if 3scale is enabled in 2.1

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Comment by luksa** [comment]
Can't add unit test, because that would introduce an import cycle. We first need to reorganize the packages. The main problem is that `Version` returns the strategy used for validation and a bunch of other stuff. This needs to be uncoupled from the version.

File: pkg/controller/versions/strategy_v2_1.go [hunk 1]
```diff
@@ -181,6 +182,17 @@ func (v *versionStrategyV2_1) validateMixerDisabled(spec *v2.ControlPlaneSpec, a
 	return allErrors
 }
 
+func (v *versionStrategyV2_1) validateAddons(spec *v2.ControlPlaneSpec, allErrors []error) []error {
+	if spec.Addons == nil {
+		return allErrors
+	}
+
+	if spec.Addons.ThreeScale != nil && spec.Addons.ThreeScale.Enabled != nil && *spec.Addons.ThreeScale.Enabled {
+		allErrors = append(allErrors, fmt.Errorf("support for 3scale has been removed in v2.1; please configure the 3scale WebAssembly adapter using a ServiceMeshExtension resource"))
```
Comment (jwendell) [inline pkg/controller/versions/strategy_v2_1.go]: Should we also explicitly instruct the user to remove the 3scale block?

Comment (dgn) [reply pkg/controller/versions/strategy_v2_1.go]: +1

Comment (luksa) [reply pkg/controller/versions/strategy_v2_1.go]: Done.

**Review by dgn — APPROVED** [review]

**Review by jwendell — APPROVED** [review]

**Pull request merged**